v1/ast/term.go

```diff
@@ -2697,31 +2691,28 @@ func (obj *object) insert(k, v *Term, resetSortGuard bool) {
 
 	for curr := head; curr != nil; curr = curr.next {
 		if equal(curr.key.Value) {
-			// The ground bit of the value may change in
-			// replace, hence adjust the counter per old
-			// and new value.
-
 			if curr.value.IsGround() {
 				obj.ground--
 			}
 			if v.IsGround() {
 				obj.ground++
 			}
 
+			// Update hash based on the new value
 			curr.value = v
+			obj.elems[hash] = curr
+			obj.hash = 0
+			for hash := range obj.elems {
```

Comment:
This is shadowing the `hash` var up top. This shouldn't cause too much confusion, but consider renaming.

Reply:
Done